test/app/test/core.clj

```diff
@@ -148,6 +153,56 @@
   (let [cl (parseType "ClassWithCallToFinalizeWithParams")]
     (is (false? (calls-finalizers? cl)))))
 
+(deftest testClassThatOverridesToString
+  (let [type-solver-classes (flatten (map allTypes sampleClassesItem10Test))
+        cl (first (filter #(= (.getName %) "ClassThatOverridesToString")
+                          type-solver-classes))]
+    (is (false? (does-not-override-toString-but-should?
+                 type-solver-classes cl)))))
+
+(deftest testClassWhoseParentsOverrideToString
+  (let [type-solver-classes (flatten (map allTypes sampleClassesItem10Test))
+        cl (first (filter #(= (.getName %) "ClassWhoseParentOverridesToString")
+                          type-solver-classes))]
+    (is (false? (does-not-override-toString-but-should?
+                 type-solver-classes cl)))))
+
+(deftest testClassThatDeclaresToStringWithParams
+  (let [type-solver-classes (flatten (map allTypes sampleClassesItem10Test))
+        cl (first (filter
+                   #(= (.getName %) "ClassThatDeclaresToStringWithParams")
+                   type-solver-classes))]
+    (is (does-not-override-toString-but-should? type-solver-classes cl))))
+
+(deftest testClassThatDoesNotOverrideToString
+  (let [type-solver-classes (flatten (map allTypes sampleClassesItem10Test))
+        cl (first (filter #(= (.getName %) "ClassThatDoesNotOverrideToString")
+                          type-solver-classes))]
+    (is (does-not-override-toString-but-should? type-solver-classes cl))))
+
+(deftest testUtilClassThatDoesNotOverrideToString
+  (let [type-solver-classes (flatten (map allTypes sampleClassesItem10Test))
+        cl (first (filter #(= (.getName %) "UtilsClassDoesNotOverrideToString")
+                          type-solver-classes))]
+    (is (false? (does-not-override-toString-but-should?
+                 type-solver-classes cl)))))
+
+(deftest test-getAllSuperclasses-depth-0
```

Comment:
...and I should have added a comment for introducing this and the following test :)